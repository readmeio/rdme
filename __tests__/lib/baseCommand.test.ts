import type { CommandClass } from '../../src/index.js';

import { beforeEach, describe, expect, it } from 'vitest';

import BaseCommand from '../../src/lib/baseCommand.js';
import configStore from '../../src/lib/configstore.js';
import { type OclifOutput, runCommand } from '../helpers/oclif.js';

class TestCommand extends BaseCommand<typeof BaseCommand> {
  async run() {
    this.info('Running test command');

    const result = await this.runRdmeCommand('whoami');

    return result;
  }
}

describe('BaseCommand', () => {
  let run: (args?: string[]) => OclifOutput;

  describe('#runRdmeCommand', () => {
    beforeEach(async () => {
      configStore.clear();
      run = runCommand(TestCommand as CommandClass);
    });

    it('should run another rdme command', async () => {
      const result = await run();

      expect(result).toMatchSnapshot();
    });
  });
});
