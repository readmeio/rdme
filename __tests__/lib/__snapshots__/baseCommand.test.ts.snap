// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseCommand > #runRdmeCommand > should run another rdme command 1`] = `
{
  "error": [Error: Please login using \`rdme login\`.],
  "stderr": "",
  "stdout": "ℹ️  Running test command
",
}
`;
